src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1,7 +1,7 @@
-//! Implementation of the test-related targets of the build system.
+//! Build-and-run steps for `./x.py test` test fixtures
 //!
-//! This file implements the various regression test suites that we execute on
-//! our CI.
+//! `./x.py test` (aka [`Kind::Test`]) is currently allowed to reach build steps in other modules.
```

Comment:
i wonder if it makes sense to link to src/bootstrap/README and/or https://rustc-dev-guide.rust-lang.org/building/bootstrapping/how-bootstrap-does-it.html in lib.rs and builder.rs respectively.

Reply:
Because most of the things run when you execute `./x.py {command}` are in fact in the given `{command}.rs`, so the fact that this one has an exception stood out to me:

https://github.com/rust-lang/rust/blob/f22a0c2d9f9cd83dfe84b65e9fa1b3fa8303c7b5/src/bootstrap/src/core/builder.rs#L768-L770